lib/src/git_subprocess.rs

```diff
@@ -106,6 +106,20 @@ impl<'a> GitSubprocessContext<'a> {
             git_cmd.creation_flags(CREATE_NO_WINDOW);
         }
 
+        // The gitconfig-controlled automated spawning of the macOS `fsmonitor--daemon`
+        // can cause strange behavior with certain subprocess operations.
+        // For example: https://github.com/jj-vcs/jj/issues/6440.
+        //
+        // Nothing we're doing in `jj` interacts with this daemon, so we force the
+        // config to be false for subprocess operations in order to avoid these
+        // interactions.
+        //
+        // In a colocated repo, the daemon will still get started the first time a `git`
+        // command is run manually if the gitconfigs are set up that way.
+        #[cfg(target_os = "macos")]
```

Comment:
Remove `cfg()`? As we discussed, it doesn't make sense to spawn fsmonitor by `jj git fetch`/`clone`.

(fwiw, `if cfg!(..)` is preferred as long as the code compiles on all platforms.)

Reply:
My thought process was to limit the change in behavior to macOS since it's only an issue on macOS, but to your point, it's probably both fine and sensical to avoid the complexity of platform-based branch.